tools/src/main/scala/scala/scalanative/nir/parser/Base.scala

```diff
@@ -72,8 +72,10 @@ object Base {
   val Int: P[Int]         = neg((HexNum | DecNum).!) map (_.toInt)
   val Long: P[Long]       = neg((HexNum | DecNum).!) map (_.toLong)
   val Infinity: P[String] = P("Infinityf".!) map (_.init)
-  val Float: P[Float]     = neg(Infinity | Literals.Literals.Float.!) map (_.toFloat)
-  val Double: P[Double]   = neg(Infinity | Literals.Literals.Float.!) map (_.toDouble)
+  val Float
+    : P[Float] = neg(Infinity | Literals.Literals.Float.!) map (_.toFloat)
+  val Double
+    : P[Double] = neg(Infinity | Literals.Literals.Float.!) map (_.toDouble)
```

Comment:
Hmmm 🤔 the penalty for breaking before `:` is very high, but in this case it's not allowed to break after `=` because  `map` is an infix operator. It doesn't break on `=` when the rhs body is an infix expression because I hit on funky routing/akka DSLs where it clearly wasn't supposed to break after `=`. There is a rewrite http://scalameta.org/scalafmt/#AvoidInfix that converts misuse of infix into regular application, that should fix this particular case. 

Ideally, scalafmt should take way more control over infix operators, but I never managed to get it right for corner cases.